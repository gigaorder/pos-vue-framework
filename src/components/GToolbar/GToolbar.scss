.g-toolbar {
  display: block;
  flex: 1 1 auto;
  max-width: 100%;
  contain: layout;
  border-radius: 0 0 16px 16px;

  &-content,
  &-extension {
    display: flex;
    align-items: center;
    position: relative;
    z-index: 0;
    padding: 8px 16px;
    font-size: 20px;
    font-weight: 400;
    line-height: 1.5;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;


    > .g-btn {
      height: 48px !important;

      &.g-btn-icon {
        width: 48px !important;
        min-width: 48px;
      }

      &:first-child {
        margin-left: -12px;

        + *:not(button) {
          padding-left: 20px;
        }
      }

      &:last-child {
        margin-right: -12px;
      }
    }

    > .g-tabs {
      height: inherit;
      margin-top: -4px;
      margin-bottom: -4px;

      .g-tab {
        height: inherit;

        &:first-child {
          margin-left: -16px;
        }

        &:last-child {
          margin-right: -16px;
        }
      }
    }
  }

  &-background {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    z-index: 0;
    contain: strict;
    border-radius: inherit;

    .g-image {
      border-radius: inherit;
    }
  }

  &__absolute {
    position: absolute;
    top: 0;
    z-index: 1;
    width: 100%;
  }

  &__bottom {
    position: absolute;
    top: initial;
    bottom: 0;
    width: 100%;
    border-radius: 16px 16px 0 0;
  }

  &__dense {
    .g-toolbar-content {
      padding-top: 0;
      padding-bottom: 0;
    }
  }

  &__prominent {
    .g-btn {
      align-self: flex-start;
    }

    .g-toolbar-content {
      align-items: flex-end;
    }
  }

  &__floating {
    .g-toolbar-content {
      display: inline-flex;
    }
  }

  &__tile {
    border-radius: 0;
  }

  &__collapse {
    max-width: 112px;
    overflow: hidden;

    .g-toolbar-content > span,
    .g-toolbar-extension {
      display: none;
    }
  }
}