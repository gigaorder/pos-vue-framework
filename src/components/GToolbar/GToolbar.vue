<template>
	<div :class="classes" :style="totalHeightStyles">
		<div class="g-toolbar-background" :style="totalHeightStyles">
			<div :style="backgroundStyles"></div>
		</div>
		<div class="g-toolbar-content" :style="contentStyles">
			<slot></slot>
		</div>
		<div v-if="extended" class="g-toolbar-extension" :style="extensionStyles" ref="extension">
			<slot name="extension"></slot>
		</div>
	</div>
</template>

<script>
	import { computed, nextTick, ref } from 'vue';
  import { convertToUnit } from '../../utils/helpers';
  import { isCssColor } from '../../mixins/colorable';
  import { linearGradient } from '../../utils/colors';

  export default {
    name: 'GToolbar',
    props: {
      absolute: Boolean,
      bottom: Boolean,
      collapse: Boolean,
      color: {
        type: String,
				default: 'white'
			},
			gradient: String,
			gradientAngle: {
        type: [String, Number],
				default: 45
			},
      dense: Boolean,
      elevation: {
        type: [String, Number],
        default: 4
      },
      extended: Boolean,
      extensionHeight: {
        type: [String, Number],
				default: 'auto'
      },
      sticky: Boolean,
      flat: Boolean,
      floating: Boolean,
      prominent: Boolean,
      short: Boolean,
      src: String,
      tile: {
        type: Boolean,
        default: true
      },
      height: String,
			filled: Boolean,
    },
    setup(props) {
      const extension = ref(null)

      const classes = computed(() => ({
        'g-toolbar': true,
        ['elevation-' + (props.flat ? '0' : props.elevation)]: true,
        'g-toolbar__absolute': props.absolute,
        'g-toolbar__bottom': props.bottom,
        'g-toolbar__dense': props.dense,
        'g-toolbar__prominent': props.prominent,
        'g-toolbar__floating': props.floating,
        'g-toolbar__tile': props.tile,
        'g-toolbar__collapse': props.collapse,
        'g-toolbar__filled': props.filled,
        'g-toolbar__sticky': props.sticky,
				['bg-'+props.color.split(' ').join('-')]: props.color && !isCssColor(props.color),
      }));

      const contentHeight = computed(() => {
        if (props.height) return props.height;
        if (props.prominent && props.dense) return 96;
        if (props.prominent && props.short) return 112;
        if (props.prominent) return 128;
        if (props.dense) return 48;
        if (props.short) return 56;
        return 64
      });

			const computedExtensionHeight = computed(() => {
				nextTick(() => {
					if (extension.value) {
						if (!isNaN(parseInt(props.extensionHeight))) return props.extensionHeight
						return extension.offsetHeight
					}
					return 0
				})
			})

      const totalHeight = computed(() => {
        let contentHeightString = !isNaN(contentHeight.value) ? `${contentHeight.value}px` : contentHeight.value
        return `calc(${contentHeightString} + ${props.extended ? parseInt(computedExtensionHeight.value) : '0px'})`
      });

      const contentStyles = computed(() => ({
        'height': convertToUnit(contentHeight.value)
      }));

      const totalHeightStyles = computed(() => ({
        'height': convertToUnit(totalHeight.value)
      }));

      const extensionStyles = computed(() => ({
        'height': convertToUnit(computedExtensionHeight.value)
      }));

      const backgroundStyles = computed(() => ({
        'position': 'absolute',
        'top': '0',
        'left': '0',
        'z-index': '-1',
        'width': '100%',
        'height': convertToUnit(totalHeight.value),
				... props.src && {
          'background-image': 'url("' + props.src + '")',
          'background-position': 'center center',
          'background-size': 'cover',
				},
				... isCssColor(props.color) && {
          'background-color': [props.color]
				},
				... props.gradient && {
          'background-image': [linearGradient(props.gradient.split(','), props.gradientAngle)]
				}
      }));

      return {
        classes,
        contentStyles,
        totalHeightStyles,
        extensionStyles,
        backgroundStyles,
				computedExtensionHeight,
        extension
      }
    }
  }
</script>

<style scoped lang="scss">
	@import "GToolbar";
</style>
