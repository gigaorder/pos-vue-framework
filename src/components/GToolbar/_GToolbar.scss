.g-toolbar {
  display: block;
  flex: 1 1 auto;
  max-width: 100%;
  contain: layout;
  border-radius: 0 0 16px 16px;

  &-content,
  &-extension {
    display: flex;
    align-items: center;
    position: relative;
    z-index: 0;
    padding: 8px 16px;
    font-size: 20px;
    font-weight: 400;
    line-height: 1.5;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    ::v-deep .g-btn {

      &:hover {
        background-color: rgba(0, 0, 0, 0.1);
      }
    }

    > ::v-deep .g-tabs {
      height: inherit;
      margin-top: -4px;
      margin-bottom: -4px;

      ::v-deep .g-tab {
        height: inherit;

        &:first-child {
          margin-left: -16px;
        }

        &:last-child {
          margin-right: -16px;
        }
      }
    }
  }

  &-extension {
    padding: 0;
  }

  &-background {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    z-index: 0;
    contain: strict;
    border-radius: inherit;

    ::v-deep .g-image {
      border-radius: inherit;
    }
  }

  &__absolute {
    position: absolute;
    top: 0;
    z-index: 1000;
    width: 100%;
  }

  &__bottom {
    position: absolute;
    top: initial;
    bottom: 0;
    width: 100%;
    border-radius: 16px 16px 0 0;
  }

  &__dense {
    .g-toolbar-content {
      padding-top: 0;
      padding-bottom: 0;
    }
  }

  &__prominent {
    ::v-deep .g-btn {
      align-self: flex-start;
    }

    .g-toolbar-content {
      align-items: flex-end;
    }
  }

  &__floating {
    .g-toolbar-content {
      display: inline-flex;
    }
  }

  &__tile {
    border-radius: 0;
  }

  &__collapse {
    max-width: 112px;
    overflow: hidden;

    .g-toolbar-content > span,
    .g-toolbar-extension {
      display: none;
    }
  }

  //custom materialize search bar in toolbar
  ::v-deep .g-tf-wrapper {
    margin: 0;
    background-color: rgba(230, 230, 230, 0.3);
    color: white;
    border-radius: 4px;

    .g-tf-input,
    .g-tf-input::placeholder {
      color: inherit;
      background-color: inherit;
    }

    &:hover {
      background-color: rgba(230, 230, 230, 0.12);
    }

    .g-tf-prepend__inner {
      margin: 8px;
    }

    .g-tf-prepend__outer,
    .g-tf-append__outer {
      margin: 0;
    }
  }

  &__filled {
    .g-toolbar-content,
    .g-toolbar-extension {
      > ::v-deep .g-btn {
        &:first-child {
          margin-left: -12px;

          + *:not(button) {
            padding-left: 20px;
          }
        }

        &:last-child {
          margin-right: -12px;
        }
      }
    }
  }
}
