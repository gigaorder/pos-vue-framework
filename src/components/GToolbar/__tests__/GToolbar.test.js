import { createLocalVue, mount } from '@vue/test-utils';
import VueCompositionApi from '@vue/composition-api';
import GToolbar from '../GToolbar';

const localVue = createLocalVue()
localVue.use(VueCompositionApi)

describe('Toolbar', () => {
  let mountFunction

  beforeEach(() => {
    mountFunction = (options = {}) => {
      return mount(GToolbar, {
        localVue,
        ...options
      })
    }
  })

  it('should render', () => {
    const wrapper = mountFunction()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should has tool-bar class', function () {
    const wrapper = mountFunction()
    expect(wrapper.contains('tool-bar'))
  });
})