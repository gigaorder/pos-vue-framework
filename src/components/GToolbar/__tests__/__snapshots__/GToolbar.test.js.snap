// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render 1`] = `"<div class=\\"bottom bg-grey-lighten-3 tool-bar\\"></div>"`;
